src/vm/amd64/JitHelpers_InlineGetThread.asm

```diff
@@ -147,6 +147,11 @@ align 16
         jmp     JIT_Box
 NESTED_END JIT_BoxFastMP_InlineGetThread, _TEXT
 
+RELATIVE_PTR_LOAD macro Reg, BaseReg, Offset
```

Comment:
> the case requires runtime lookup from the generic type descriptor and signature.

This should be done via `ELEMENT_TYPE_NATIVE_ARRAY_TEMPLATE_ZAPSIG`.

> Is it good idea to add the case to `TypeHandle::IsCanonicalSubtype`?

I do not see a problem with it.